### DISCUSSION TITLE
The requireVersion or version of ModuleFederationPlugin is not work

### DISCUSSION THREAD

**Discussion by karmiy** [webpack/webpack#14388] [Q&A]
HOST:

```js
new ModuleFederationPlugin({
      name: "app1",
      shared: {
        react: {
          version: '16.8.0',
          import: "react",
          shareKey: "react",
          shareScope: "default",
          singleton: true,
        },
      },
}),
```

STANDALONE REMOTE:

```js
new ModuleFederationPlugin({
      name: "app2",
      shared: {
        react: {
          requiredVersion: '16.14.0',
          import: "react",
          shareKey: "react",
          shareScope: "default",
          singleton: true,
        },
      },
}),
```

I think App2 should not share with App1's React due to version mismatch. However, App2 uses App1's React

Debugging shows that they all use `__webpack_require__. E ("vendors-node_modules_react_index_js")`

May I ask why they are not differentiated according to version, but all use `vendors-node_modules_react_index_js`, Not, for example, `16.8.0/vendors-node_modules_react_index_js` and `16.14.0/vendors-node_modules_react_index_js`

Thanks!

**Comment by alexander-akait** [comment]
For questions please use https://github.com/webpack/webpack/discussions, because `requiredVersion` are different
